Core/include/SceneParser.h

```diff
@@ -0,0 +1,170 @@
+#pragma once
+
+#include "GLTFLoader.h"
+#include "Components/CameraComponent.h"
+#include "Components/LightComponent.h"
+#include "Components/ScriptComponent.h"
+#include "Managers/EntityManager.h"
+
+#include <glm/glm.hpp>
+#include <glm/gtc/type_ptr.hpp>
+#include <json.hpp>
+#include <fstream>
+#include <iostream>
+#include <unordered_map>
+
+namespace SceneParser
+{
+    using json = nlohmann::json;
+    enum SceneElement
+    {
+        assets,
```

Comment:
Minor Style, at EA/ BBI we set top enum = 0 explicitly each time

Reply:
Thinking back, enums in this fashion is not necessary for loading since it should be done in a specific order. 
Maybe useful for serializing.